go/nogo.rst

```diff
@@ -0,0 +1,307 @@
+        visibility = ["//visibility:public"],
+    )
+
+The `nogo`_ rule generates a program that analyzes Go source code. This program
+is run alongside the compiler. You must define a `nogo`_ target whose ``deps``
+attribute contains all analyzer targets. These analyzers will be linked to the
+generated ``nogo`` binary and executed at build-time.
+
+.. code:: bzl
+
+    load("@io_bazel_rules_go//go:def.bzl", "nogo")
+
+    nogo(
+        name = "nogo",
+        deps = [
+            ":importunsafe",
+            ":unsafedom",
+            "@analyzers//:loopclosure", # analyzers can be imported from a remote repo
+        ],
+        visibility = ["//visibility:public"], # must have public visibility
+    )
+
+**NOTE**: Writing each ``nogo`` analyzer as a `go_tool_library`_ rule instead of
+a `go_library`_ rule avoids a circular dependency: `go_library`_ implicitly
+depends on `nogo`_, which depends on analyzer libraries, which must not depend
+on `nogo`_. `go_tool_library`_ does not have the same implicit dependency.
+
+Finally, the `nogo`_ target must be passed to ``go_register_toolchains``
+in your ``WORKSPACE`` file.
+
+.. code:: bzl
+
+    load("@io_bazel_rules_go//go:def.bzl", "go_rules_dependencies", "go_register_toolchains")
+    go_rules_dependencies()
+    go_register_toolchains(nogo="@//:nogo")
+
+The generated ``nogo`` program will run alongside the compiler when building any
+Go target (e.g. `go_library`_) within your workspace, even if the target is
+imported from an external repository. However, ``nogo`` will not run when
+targets from the current repository are imported into other workspaces and built
+there.
+
+Configuring analyzers
+~~~~~~~~~~~~~~~~~~~~~
+
+By default, ``nogo`` analyzers apply to all Go source files being compiled. This
+behavior can be changed with a JSON configuration file.
+
+The top-level JSON object in the file must be keyed by the name of the analyzer
+being configured. These names must match the ``Analysis.Name`` of the registered
+analysis package. The JSON object's values are themselves objects which may
+contain the following key-value pairs:
+
++----------------------------+---------------------------------------------------------------------+
+| **Key**                    | **Type**                                                            |
++----------------------------+---------------------------------------------------------------------+
+| ``"description"``          | :type:`string`                                                      |
++----------------------------+---------------------------------------------------------------------+
+| Description of this analyzer configuration.                                                      |
++----------------------------+---------------------------------------------------------------------+
+| ``"apply_to"``             | :type:`dictionary, string to string`                                |
```

Comment:
I agree this should be changed. Actually, I think `whitelist` should be changed, too. It would be good to have these obviously be a pair so it's clear from the names this is an include / exclude system. It would also be good to clarify these operate on file names, not e.g., function names.

So maybe `only_files` and `exclude_files`?

+1 to a separate commit, but today if possible. I'd like to tag a release tomorrow (but I'll hold the release for this if we need to).